models/sipnet/R/write.configs.SIPNET.R

```diff
@@ -186,78 +186,85 @@ write.config.SIPNET <- function(defaults, trait.values, settings, run.id, inputs
     
     # Remove NAs. Constants may be specified as NA to request template defaults. Note that it is 'NA'
     # (character) not actual NA due to being read in as XML
-    pft.names <- pft.names[pft.traits != "NA" & !is.na(pft.traits)]
+    pft.trait.names <- pft.trait.names[pft.traits != "NA" & !is.na(pft.traits)]
     pft.traits <- pft.traits[pft.traits != "NA" & !is.na(pft.traits)]
     pft.traits <- as.numeric(pft.traits)
     
     # Leaf carbon concentration
-    leafC <- 0.48  #0.5
-    if ("leafC" %in% pft.names) {
-      leafC <- pft.traits[which(pft.names == "leafC")]
+    leafC <- NA
+    if ("leafC" %in% pft.trait.names) {
+      leafC <- pft.traits[which(pft.trait.names == "leafC")]
       id <- which(param[, 1] == "cFracLeaf")
-      param[id, 2] <- leafC * 0.01  # convert to percentage from 0 to 1
+      param[id, 2] <- PEcAn.utils::ud_convert(leafC, "percent", "1") # Convert from percentage to fraction
+    } else {
+      leafC <- 0.48 # Fixed value if not available, because it is used in downstream calculations
     }
     
     # Specific leaf area converted to SLW
+    # leafCSpWt [gC/m2 leaf], SLA [m2 leaf/kg C], leafC [percentage C]
     SLA <- NA
     id <- which(param[, 1] == "leafCSpWt")
-    if ("SLA" %in% pft.names) {
-      SLA <- pft.traits[which(pft.names == "SLA")]
-      param[id, 2] <- 1000 * leafC * 0.01 / SLA
+    if ("SLA" %in% pft.trait.names) {
+      SLA <- pft.traits[which(pft.trait.names == "SLA")]
+      param[id, 2] <- PEcAn.utils::ud_convert(leafC / SLA, "kg/m2", "g/m2")
     } else {
-      SLA <- 1000 * leafC / param[id, 2]
+      SLA <- PEcAn.utils::ud_convert(leafC / param[id, 2], "kg", "g")
     }
     
     # Maximum photosynthesis
-    Amax <- NA
+    # SIPNET: aMax [nmol CO2 / g   leaf / sec]
+    # PEcAn:  Amax [umol CO2 / m^2 leaf / sec]
     id <- which(param[, 1] == "aMax")
-    if ("Amax" %in% pft.names) {
-      Amax <- pft.traits[which(pft.names == "Amax")]
-      param[id, 2] <- Amax * SLA
+    SLA_g <- PEcAn.utils::ud_convert(SLA, "1/kg", "1/g") 
+    if ("Amax" %in% pft.trait.names) {
+      Amax_area <- pft.traits[which(pft.trait.names == "Amax")] # [µmol/m2/s]
+      param[id, 2] <- PEcAn.utils::ud_convert(Amax_area * SLA_g, "umol", "nmol") # [nmol/g/s]
     } else {
-      Amax <- param[id, 2] * SLA
+      amax_mass <- param[id, 2] # [nmol/g/s]
+      Amax <- PEcAn.utils::ud_convert(amax_mass / SLA_g, "nmol", "umol") # [umol/m2/s]
```

Comment:
Needed for use below in response calculation, right? 
```suggestion
      Amax_area <- PEcAn.utils::ud_convert(amax_mass / SLA_g, "nmol", "umol") # [umol/m2/s]
```

Reply:
LGTM!